server/src/main/kotlin/org/javacs/kt/externalsources/KlsURI.kt

```diff
@@ -82,3 +112,25 @@ data class KlsURI(val uri: URI) {
 
     override fun toString(): String = uri.toString()
 }
+
+private fun parseKlsURIFileName(uri: URI): URI = URI(uri.toString().split("?")[0])
```

Comment:
What do you think about calling it `parseKlsURIFileURI`, since the result includes the entire path/URI, not just the name?

Reply:
Yep, makes sense